clients/lsp-lua.el

```diff
@@ -370,6 +378,24 @@ and `../lib` ,exclude `../lib/temp`.
    ("Lua.completion.callSnippet" lsp-lua-completion-call-snippet)
    ("Lua.color.mode" lsp-lua-color-mode)))
 
+(defun lsp-lua-language-server-install (client callback error-callback update?)
+  "Download the latest version of lua-language-server and extract it to
+`lsp-lua-language-server-install-dir'."
+  (ignore client update?)
+  (let ((tmp-zip (make-temp-file "ext" nil ".zip")))
+    (delete-file tmp-zip)
+    (lsp-download-install
```

Comment:
Something like
```emacs-lisp
(lsp-download-install (lambda (&rest _) (set-file-mode lsp-clients-lua-language-server-bin #o0700) (funcall callback)) error-callback :url (lsp-vscode-extension-url "sumneko" "lua" "1.17.0" :store-path lsp-clients-lua-language-server-install-dir :decompress :zip)

Reply:
If you don't mind I propose merging as it is and later addressing both lsp-eslint.el and lsp-lua.el.

Reply:
Totally 👍

Reply:
@nbfalcon The bin for lua-language-sever is in the subtree of the folder structure, while `lsp-download-install` is not the best choice.

Reply:
@sunlin7 I don't quite understand - `lsp-download-install` is downloading a zip, which is unzipped to a directory, so this shouldn't be a problem. In any case, I agree with @yyoncho that this can be addressed later.